base/reducedim.jl

```diff
@@ -3,7 +3,7 @@
 ## Functions to compute the reduced shape
 
 # for reductions that expand 0 dims to 1
-reduced_index(i::OneTo) = OneTo(1)
+reduced_index(i::OneTo) = typeof(i)(1)
```

Comment:
Just seems a little bit easier to read to me this way:
```suggestion
reduced_index(i::OneTo{T}) where {T} = OneTo(one(T))
```